### PR TITLE
Basics of Convolution Neural Networks (CNNs)

### LINKED ISSUE #493: Basics of Convolution Neural Networks (CNNs)
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

Convolutional Neural Networks (CNN) is a technique in deep neural networks commonly applied to analyzing visual images. 
CNNs have been applied in:
1. Image and Video recognition.
2. Image classification.
3. Natural Language Processing
4. Medical image analysis.
5. Recommender Systems

### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

By the end of the article, the reader should be able to understand:
1. The architecture of Convolutional Neural Networks.
2. Convolutional Neural Networks Optimization methods.
3. Applications of Convolution Neural Networks

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

1. Murphy, J. (2016). An overview of convolutional neural network architectures for deep learning. Microway Inc.
2. Bambharolia, P. (2017). Overview of Convolutional Neural Networks. In Conference: International Conference on Academic Research in Engineering and Management (pp. 25-31).

### PR DESCRIPTION
**Pull request by Bayler**
Fixed #493

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by Bayler** [comment]
This Fixes #493

**Comment by hectorkambow** [comment]
Plagiarism check done

File: articles/basics-of-convolution-neural-networks/index.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+### Terminologies
+
+1. Parameter
+
+These are placeholders that change during training. Weights and Biases are the two main parameters in CNNs.
+
+2. Hyperparameter
+
+A hyperparameter is also a placeholder, but unlike the parameter, hyperparameters need to set before training starts. The learning rate is one of the important hyperparameters that need to be set before training starts.
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:28]: It is important to specify that hyper-parameters can be modified as the model learns.

File: articles/basics-of-convolution-neural-networks/index.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+1. Parameter
+
+These are placeholders that change during training. Weights and Biases are the two main parameters in CNNs.
+
+2. Hyperparameter
+
+A hyperparameter is also a placeholder, but unlike the parameter, hyperparameters need to set before training starts. The learning rate is one of the important hyperparameters that need to be set before training starts. 
+
+3. Kernel
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:30]: This definition seems to be a bit off. Kernels are learned during the training process. Convolution is an operation performed on the image with the kernel in the first layer. The convolved image is convolved with kernels in the second layer, and so on.  

You mention learnable parameters. How are they different from the first point?

File: articles/basics-of-convolution-neural-networks/index.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+A kernel is an integral component of the structure of a CNN. It consists of a set of learnable parameters that are applied to the operations of convolutions. 
+
+4. Stride
+
+Stride refers to the number of pixels a convolution filter moves. It can be compared to a sliding window. The image below shows the stride, S moving with a pixel of 2
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:36]: I don't think it is wise to mention the convolution filter initially. Explain convolution first, and then move onto these definitions.

File: articles/basics-of-convolution-neural-networks/index.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+![Stride, S](/engineering-education/basics-of-convolution-neural-networks/stride.PNG)<br>
+
+*[Image Source: Stanford](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-convolutional-neural-networks)*
+
+### What is Convolution Neural Networks (CNN)?
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:42]: What is -> What  are

File: articles/basics-of-convolution-neural-networks/index.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+Convolution Neural Networks (CNN) is a class of deep learning techniques popularly use to solve computer vision tasks.
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:44]: Convolution Neural Networks (CNNs) are a

File: articles/basics-of-convolution-neural-networks/index.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+CNN learns the spatial features of inputs through a process known as backpropagation, as we will see in the image below. Backpropagation can be described as the backward propagation of errors. It is the core algorithm behind how neural networks learn. Learn more about it in this [video](/https://www.youtube.com/watch?v=Ilg3gGewQ5U/).
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:45]: Factual error: CNN learns the spatial features through convolution operation. Backpropagation is used to update the weights and minimize/maximize the objective function.

File: articles/basics-of-convolution-neural-networks/index.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+The CNN architecture consists of a stacking of three building blocks:
+
+1. Convolution layers
+
+2. Pooling layers, i.e., max pooling
+
+3. Fully connected (FC) layers
+
+#### Convolution layer
+
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:64]: Mention the convolution operation. It is essential to explain the feature extraction part.

File: articles/basics-of-convolution-neural-networks/index.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+Mathematically, this is defined by: 
+
+f(x) = max (0, x)
+
+![Rectified Linear Unit](/engineering-education/basics-of-convolution-neural-networks/relu-function.PNG)<br>
+
+*[Image Source: Papers with code](https://paperswithcode.com/method/relu)*
+
+ReLU is the default activation function when working with CNNs, as it tends to have a faster training time.
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:81]: It is widely used but need not be the default one.

File: articles/basics-of-convolution-neural-networks/index.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+ReLU is the default activation function when working with CNNs, as it tends to have a faster training time. 
+
+Other activation functions used in neural networks include the sigmoid and the hyperbolic tangent. Read more about them [here](/https://en.wikipedia.org/wiki/Activation_function/).
+
+#### Pooling layer
+
+A pooling layer is a form of non-linear downsampling that reduces the dimensionality of the feature maps of an input image, making its representation smaller and more manageable. This process reduces the number of parameters and computation resources in the network.
+
+It is important to note that a 2 x 2 filter size applied with a stride of 2 is the most used in CNN.
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:89]: Take an example and explain the same. What are the initial dimensions, and how they are reduced using pooling operation?

File: articles/basics-of-convolution-neural-networks/index.md [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+*[Image Source: Stanford](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-convolutional-neural-networks)*
+
+### Applications of Convolution Neural Networks
+
+#### Computer Vision
+
+Image classification is a very common problem in computer vision. The goal in image classification is to learn features directly from image data. These learned features are then used for classification. For instance, feeding an image of a cat as our input, the network should classify the image as a cat. 
+
+Recently, CNN has been used for image captioning. Here, the goal is to generate a sentence that describes the semantic content in an image. It is achieved by combining CNN with a [Recurrent Neural Network](/https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-recurrent-neural-networks/)(RNN). In the last layer in the CNN architecture, instead of a fully connected (FC) layer, the last layer of the CNN is replaced with an RNN.
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:122]: I feel some more explanation is required here. A beginner will find it difficult to understand how an RNN is getting fixed to a CNN.

File: articles/basics-of-convolution-neural-networks/index.md [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+
+Recently, CNN has been used for image captioning. Here, the goal is to generate a sentence that describes the semantic content in an image. It is achieved by combining CNN with a [Recurrent Neural Network](/https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-recurrent-neural-networks/)(RNN). In the last layer in the CNN architecture, instead of a fully connected (FC) layer, the last layer of the CNN is replaced with an RNN.   
+
+#### Natural Language Processing (NLP)
+
+Traditionally, CNNs have only been applied in solving Computer Vision problems. This has changed over the years. CNN's are now being applied to solve NLP problems such as speech recognition and text classification. 
+The rapid development of speech recognition systems has been on the increase over the years. Apple's Siri and Amazon's Alexa are a few examples of such systems available today.
+As for text classification, texts are first converted into vectors. These vectors are then passed as inputs to the CNN. In a forward propagating manner, these vectors pass through the convolution and pooling layer to extract features and reduce dimensions.
```
Comment (lalith1403) [inline articles/basics-of-convolution-neural-networks/index.md:128]: Mention word vectors.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
@Bayler, I feel a lot of content is missing. An introductory article on CNNs needs to go in-depth into the workings of the convolution operation. Moreover, by reading this, the person should understand what CNN is. Currently, the structure of the article is skewed towards ensuring all definitions are included. 

It would be best if you grasped the attention of the reader. Why do we need CNNs?
How are CNNs better than other algorithms? And then start with neural networks and convolution operation. Then strides, pooling, etc. Finally, end at using various optimizers to update the network to minimize/maximize objective accordingly.

The applications for NLP and image captioning need to be explained in depth.

**Comment by Bayler** [comment]
Noted @lalith1403 I'll make those revisions

**Review by lalith1403 — APPROVED** [review]

**Pull request merged**